content/blog/survival-analysis-parsnip-adjacent/index.Rmd

```diff
@@ -0,0 +1,227 @@
+
+```{r new-mode}
+library(censored)
+
+decision_tree() %>% 
+  set_engine("rpart") %>% 
+  set_mode("censored regression")
+
+```
+
+You may notice a new package name here, `censored`. The number of models supported in tidymodels is growing. The core models live in parsnip, however there are several parsnip-adjacent extension packages with additional models such as [discrim](https://discrim.tidymodels.org/) for discriminant analysis models, [poissonreg](https://poissonreg.tidymodels.org/) for Poisson regression models, and [baguette](https://baguette.tidymodels.org/) for ensemble models via bagging. The [censored](https://tidymodels.github.io/censored/) package is the extension package for censored regression. 
+
+Previously we kept the code for the model specifications and their engines in the relevant extension package. While this looks like a straightforward solution to keep code modular and maintainable, we are now reaching such richness in available models that we ran into the following: If the model specification for a `bag_tree()` lives in baguette but we want to add an engine for the new "censored regression" mode in censored, censored has to depend on both parsnip and baguette. To avoid such dependency structures, we are now moving towards keeping all model specifications in parsnip and the extensions packages contain the engine code. 
+
+
+### Model fitting
+
+For fitting a model, parsnip typically supports a formula and a matrix interface, regardless of the interface of the engine. For censored regression models, we are currently only supporting a formula interface. Most engines already support this type of interface, and a formula allows us to specify the outcome via the standard `survival::Surv()` function, tying together event time and status. 
+
+
+A formula also let's us specify stratification for a proportional hazards model. For the `survival::coxph()` function, which powers the `survival` engine for `proportional_hazards()`, this may look like 
+
+<div align="center">
+  `Surv(time, event) ~ x + z + strata(s)`.
+</div>
+
+
+The glmnet package, which powers the `glmnet` engine, does not have a formula interface. It requires us to specify the strata by stratifying the response via `stratifySurv(y, s)`. The response is typically a `Surv` object, so you could end up with a formula like this:
+
+<div align="center">
+  `stratifySurv(Surv(time, event), strata) ~ x`.
+</div>
+
+This works for fitting the model but the syntax is rather convoluted on the left-hand side of the formula. The specification of the stratification is inconsistent with the approach taken by the survival package. Additionally, without further modification, prediction fails because variables on the left-hand side of the formula are treated as response variables by parsnip and not available at prediction time.
```

Comment:
```suggestion
This works for fitting the model but the syntax is rather convoluted on the left-hand side of the formula. The specification of the stratification is inconsistent with the approach taken by the survival package. Additionally, without further modification, prediction fails because variables on the left-hand side of the formula are treated as response variables by parsnip and are not available at prediction time.
```